tests/NovelRT.Tests/CMakeLists.txt

```diff
@@ -55,6 +55,44 @@ set(SOURCES
 )
 
 add_executable(Engine_Tests ${SOURCES})
+
+if(MSVC)
+  set_property(TARGET Engine_Tests PROPERTY WINDOWS_EXPORT_ALL_SYMBOLS ON)
+  target_compile_options(Engine_Tests
```

Comment:
We should likely use the same approach as in [Fabulist](https://github.com/novelrt/Fabulist/blob/main/thirdparty/lua/CMakeLists.txt#L115) here, to make switching compilers faster

Reply:
Eh?

Wouldn't this mean adjusting the core NovelRT compiler switches as well?

Reply:
We should probably be keeping things consistent.

Probably.

But that might break too much lmao. :D

Reply:
My thinking here basically is we should probably go all one way or not bother at all, this is supposed to be a low cost fix if it can be helped.

Basically.

Reply:
Something like this?

`target_compile_options(Engine_Tests`
`    PRIVATE`
`        $<$<CXX_COMPILER_ID:MSVC>:/W4>`
`        $<$<CXX_COMPILER_ID:MSVC>:/WX>`

etc.

Reply:
Yeah that looks right. :)

Reply:
> I mean, ideally, yes, but that can likely be done in a follow up PR for a cleaner changeset if that's desired.

So, should I change these here or will there be another issue for this topic?

Reply:
Change it here

Reply:
Noted. Should I change all the options definition? Like, in the Tests project but also in the Interop and others?